eng/intellisense.targets

```diff
@@ -0,0 +1,32 @@
+<Project>
+
+  <PropertyGroup>
+      <!-- Search for the documentation file in the intellisense package and otherwise pick up the generated one. -->
+    <IntellisensePackageFolderName Condition="'$(IsPackable)' == '' or '$(IsPackable)' == 'false'">net</IntellisensePackageFolderName>
+    <IntellisensePackageFolderName Condition="'$(IsPackable)' == 'true'">dotnet-plat-ext</IntellisensePackageFolderName>
+    <LibIntellisenseDocumentationFilePath Condition="'$(UseIntellisenseDocumentationFile)' == 'true'">$([MSBuild]::NormalizePath('$(ArtifactsBinDir)', '$(MSBuildProjectName)', '$(LibrariesConfiguration)', '$(NetCoreAppCurrent)', '$(MSBuildProjectName).xml'))</LibIntellisenseDocumentationFilePath>
+    <LibIntellisenseDocumentationFilePath Condition="'$(UseIntellisenseDocumentationFile)' == '' or '$(UseIntellisenseDocumentationFile)' == 'false'">$([MSBuild]::NormalizePath('$(XmlDocFileRoot)', '$(IntellisensePackageFolderName)', '1033', '$(AssemblyName).xml'))</LibIntellisenseDocumentationFilePath>
+    <UseIntellisenseDocumentationFile Condition="'$(UseIntellisenseDocumentationFile)' == '' and Exists('$(LibIntellisenseDocumentationFilePath)')">true</UseIntellisenseDocumentationFile>
+  </PropertyGroup>
+
+  <!-- TODO: Remove this target when no library relies on the intellisense documentation file anymore.-->
+  <Target Name="ChangeDocumentationFileForPackaging"
+          AfterTargets="DocumentationProjectOutputGroup"
+          Condition="'$(UseIntellisenseDocumentationFile)' == 'true'">
+    <ItemGroup>
+      <DocumentationProjectOutputGroupOutput Remove="@(DocumentationProjectOutputGroupOutput)" />
+      <DocumentationProjectOutputGroupOutput Include="$(LibIntellisenseDocumentationFilePath)" />
+    </ItemGroup>
+  </Target>
+
+  <Target Name="CopyDocumentationFileToXmlDocDir"
+          AfterTargets="CoreCompile"
+          Condition="'$(TargetFramework)' == '$(NetCoreAppCurrent)'"
```

Comment:
I think this should only run for src projects.  I think this ends up running in `ref` projects right now and doing nothing since the `DocumentationProjectOutputGroupOutput` item is empty.  You probably want to condition this to only run for src projects since you wouldn't want two projects with the same name copying xml files to a single directory.  We're also only expecting the src projects to actually contain the docs.  `IsSourceProject` should probably be considered here.

Reply:
Such condition would be: `'$(IsSourceProject)' == 'true'`, is that right?

Should I add the condition only to this target? Or should I add such condition to the whole new `intellisense.targets` import file?

I ask because in `Directory.Build.targets`, this import was added with a condition to skip pkgproj extension projects, but I could also add the src condition you're talking about there.

Reply:
You could do that, it's more of a style question.  @ViktorHofer do you have any style preferences for understanding the imports in our repo?  Do we want conditions on imports and targets, or would you prefer to just place the conditions on targets to avoid mistakes where folks assume targets are imported?

Reply:
Well, the style is already there, where we import intellisense.targets. I just extracted the intellisense-specific Targets from the old file and put it in a new one. My understanding is that the condition for importing both *.targets files should be the same, but I need clarification in this other comment: https://github.com/dotnet/runtime/pull/79134#discussion_r1068378602